### DISCUSSION TITLE
Path not being grabbed correctly by the extension and giving ENOENT

### DISCUSSION THREAD

**Discussion by asapsmc** [James-Yu/LaTeX-Workshop#3551] [Q&A]
I'm unable to use LatexWorkshop+VSCode as I'm getting the known error `LaTeX fatal error: spawn latexmk ENOENT, . PID: undefined.`. I'm using a fresh install of TexLive on Macos.
Yet, `/Library/Tex/texbin` is in my path, as you can check from the output of `echo $PATH`:
`/Users/myusername/.nvm/versions/node/v11.10.0/bin:/Users/myusername/miniconda3/condabin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/opt/X11/bin:/Users/myusername/.fig/bin`.
Here is some other relevant system info.
```
> which perl
/usr/local/bin/perl
> which tlmgr
/Library/TeX/texbin/tlmgr
> which latexmk
/Library/TeX/texbin/latexmk
```

However, when I try to compile I inevitably get the following error:
```
LaTeX build process spawned. PID: undefined.
LaTeX fatal error: spawn latexmk ENOENT, . PID: undefined.
Does the executable exist? $PATH: /usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin
Does the executable exist? $Path: undefined
The environment variable $SHELL: /bin/zsh

```

I even tried to force in LatexWorkshop settings by using the following in latexmk`"env": {
        "TEXMFHOME": "/Library/TeX/texbin/latexmk"
      }`

I also restarted VSCode and OS several times. I already checked all related issues here (e.g. [this](https://github.com/James-Yu/LaTeX-Workshop/issues/1337)), and also several issues on stackoverflow and tex.stackexchange, but I just can't make this work. 
Can anyone help me here?

**Comment by scottstanie** [comment]
I was having this issue as well, even though my first use of Latex Workshop had no issues. 
When it was erroring for me, it was printing `Does the executable exist? $PATH: /usr/bin:/usr/sbin:/sbin:` similar to you, so it was missing the right `$PATH` id' having in my terminal (or even when I opened a terminal in VSCode, I'm on a Mac).

My fix was to close VSCode, then open it from the terminal with `code .` in the directory of my latex project. The PATHs were all correctly set again.

**Reply by ghost** [reply]
It is possible that your terminal opens in administrative mode.

Openning vscode in administrative mode also works.

**Comment by if-david** [comment]
Don't waste any more time—after trying at least 10 different solutions, the only one that works is installing the Full MacTeX."

**Comment by rileyjmurray** [comment]
On macos I got things to work by navigating to my applications folder in Finder, right clicking "Visual Studio Code", and then selecting "Open". I think this gives the application slightly elevated permissions compared to clicking the icon normally.
